Adds pagination

helrond opened the pull request:
Adds search results pagination, fixes #94 and fixes #14 

I've also added some missing tests and cleaned up some unnecessary code.

helrond committed:
remove page calculation

helrond committed:
add pagination component

helrond committed:
restructuring search page HTML

helrond committed:
updating styles

helrond committed:
reset offset when facets and sort are clicked

helrond committed:
delete offset if it is zero

helrond committed:
adds proptypes and tests

helrond committed:
resolving merge conflicts

helrond committed:
add missing tests

helrond committed:
adding values back to search box and category select

HaSistrunk commented:
Cool, this works! The designs show the pagination component both above and below the search results. Are you thinking to try it above first and see how it goes?

Also, I'm not sure where this was introduced, but the auto-filled dates in the facets modal are funky. Eg. after searching "ford" the date inputs look like: 
![image](https://user-images.githubusercontent.com/15254041/93383014-06d61200-f831-11ea-8025-d9bddce740a8.png)

Finally, this is a little late, but what do you think about giving the `tile__hit-count` a hover style similar to `btn--orange`? I know the orange colors are slightly different...
```
&:hover {
    background: $med-brown;
    border: 1px solid $med-brown;
    color: $off-white;
```

helrond committed:
add background chagne on hover

helrond committed:
adds pagination to bottom of pages

helrond committed:
add transition

helrond commented:
Thanks for the catch on the pagination - had not noticed it was at the bottom too! I've added it there as well and goshdarn flexbox is awesome.

I also added the background change on hover for the search tips - good suggestion.

The weird dates are coming from indexing. I need to dig into that a bit more and figure out what's going on. There's some Elasticsearch magic that happens around dates that I need to understand better.
